src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Entity/CopyEntityCommandHandler.php

```diff
@@ -129,36 +125,28 @@ public function handle(CopyEntityCommand $command)
             );
         }
 
-        $saveEntityCommand->setStatus(Entity::STATE_PUBLISHED);
-        $saveEntityCommand->setId($dashboardId);
-        $saveEntityCommand->setService($command->getService());
-        $saveEntityCommand->setManageId($command->getManageId());
+        // Convert manage entity to domain entity
+        $domainEntity = Entity::fromManageResponse($manageEntity);
+
+        // Set some defaults
+        $domainEntity->setStatus(Entity::STATE_PUBLISHED);
+        $domainEntity->setId($dashboardId);
+        $domainEntity->setService($command->getService());
+        $domainEntity->setManageId($command->getManageId());
 
         // Published production entities must be cloned, not copied
         $isProductionClone = $command->getEnvironment() == 'production' && $manageStagingState === 0;
         // Entities copied from test to prod should not have a manage id either
         $isCopyToProduction = $command->getEnvironment() == 'production' && $command->getSourceEnvironment() == 'test';
         if ($isProductionClone || $isCopyToProduction) {
-            $saveEntityCommand->setManageId(null);
-        }
-
-        $this->commandBus->handle(
-            new LoadMetadataCommand(
```

Comment:
I was unable to find at the moment any reason why this was needed. The only thing I could find why this was needed is because manage wasn't initaly the single source of truth and therefor it was needed to update the sp dashbaords internal entity to the latest state of the entity as in manage.